che-theia-java-extension/src/browser/navigation/file-structure.ts

```diff
@@ -0,0 +1,269 @@
+}
+
+/**
+ * Object with parameters to execute custom command for getting file structure.
+ */
+export interface FileStructureCommandParameters {
+    uri: string,
+    showInherited: boolean
+}
+
+@injectable()
+export class FileStructure implements QuickOpenModel, CommandContribution, KeybindingContribution, MenuContribution {
+    /**
+     * Whether to hide members from super classes.
+     */
+    protected showInheritedMembers: boolean = false;
+    /**
+     * Whether the dialog is currently open.
+     */
+    protected isOpen: boolean = false;
+    /**
+     * The current lookFor string input by the user.
+     */
+    protected currentLookFor: string = "";
+
+    private items!: QuickOpenItem[];
+    private command: Command = {
+        id: 'java.command.file.structure',
+        label: 'Java: Navigate File Structure'
+    };
+    
+    constructor(
+        @inject(CommandRegistry) protected readonly commands: CommandRegistry,
+        @inject(KeybindingRegistry) protected readonly keybindingRegistry: KeybindingRegistry,
+        @inject(KeybindingRegistry) protected readonly keybindings: KeybindingRegistry,
+        @inject(QuickOpenService) protected readonly quickOpenService: QuickOpenService,
+        @inject(LanguageClientProvider) protected readonly languageClientProvider: LanguageClientProvider,
+        @inject(EditorManager) protected readonly editorManager: EditorManager
+    ) { 
+    }
+
+    async open () {
+        // Triggering the keyboard shortcut while the dialog is open toggles
+        // showing the inherited members.
+        if (this.isOpen) {
+            this.showInheritedMembers = !this.showInheritedMembers;
+        } else {
+            this.showInheritedMembers = false;
+            this.isOpen = true;
+        }
+
+        let placeholderText = "File Structure.";
+        const keybinding = this.getKeyCommand();
+        const showOrHide = this.showInheritedMembers ? 'hide' : 'show';
+        if (keybinding) {
+            placeholderText += ` (Press ${keybinding} to ${showOrHide} inherited members)`;
+        }
+
+        const client = await this.languageClientProvider.getLanguageClient("java");
+        if (client) {
+            const fileStructureResponse = await this.doRequestToStructure(client);
+
+            this.items = [];
+            this.combineItems(fileStructureResponse);
+            this.quickOpenService.open(this, {
+                prefix: this.currentLookFor,
+                fuzzyMatchLabel: true,
+                fuzzyMatchDescription: true,
+                fuzzySort: true,
+                placeholder: placeholderText,
+                onClose: () => {
+                    this.isOpen = false;
+                    this.currentLookFor = "";
+                },
+            });
+        }
+    }
+
+    private combineItems(members: Members[]) {
+        for (const symbolInformation of members) {
+            const info = symbolInformation.info as MemberItem;
+            if(!info){
```

Comment:
maybe some issues with style on this file (like missing space before parenthesis, etc)